server/tests/test_file_source_api.py

```diff
@@ -0,0 +1,167 @@
+from datetime import timedelta
+
+import pytest
+from django.urls import reverse
+
+from server.files.models import FileSource
+
+
+@pytest.fixture
+def test_file_source(test_notebook):
+    return FileSource.objects.create(
+        notebook=test_notebook,
+        filename="foo.csv",
+        url="https://iodide.io/foo",
+        update_interval=timedelta(days=1),
+    )
+
+
+@pytest.fixture
+def file_source_post_blob(test_notebook):
+    return {
+        "notebook_id": test_notebook.id,
+        "filename": "foo2.csv",
+        "url": "https://iodide.io/foo2",
+        "update_interval": "1 day, 0:00:00",
+    }
+
+
+@pytest.fixture
+def file_source_update_blob(test_notebook):
+    return {
+        "notebook_id": test_notebook.id,
+        "filename": "foo3.csv",
+        "url": "https://iodide.io/foo3",
+        "update_interval": "1 week, 0:00:00",
+    }
+
+
+def test_list_file_sources_for_notebook(client, test_notebook, test_file_source, fake_user):
+    client.force_login(user=fake_user)
+    resp = client.get(
+        reverse("notebook-file-sources-list", kwargs={"notebook_id": test_notebook.id})
+    )
+    assert resp.status_code == 200
+    assert resp.json() == [
+        {
+            "filename": test_file_source.filename,
+            "id": test_file_source.id,
+            "latest_file_update_operation": None,
+            "update_interval": str(test_file_source.update_interval.total_seconds()),
+            "url": test_file_source.url,
+        }
+    ]
+
+
+@pytest.mark.parametrize("logged_in", [True, False])
+def test_list_file_sources_for_notebook_unauthorized(
+    client, test_notebook, test_file_source, fake_user2, logged_in
+):
+    # two cases: logged in as notebook non-owner, not logged in
+    # in either case, it should *not* reveal the url of the file source
+    if logged_in:
+        client.force_login(user=fake_user2)
+    resp = client.get(
+        reverse("notebook-file-sources-list", kwargs={"notebook_id": test_notebook.id})
+    )
+    assert resp.status_code == 200
+    assert resp.json() == [
+        {
+            "filename": test_file_source.filename,
+            "id": test_file_source.id,
+            "latest_file_update_operation": None,
+            "update_interval": str(test_file_source.update_interval.total_seconds()),
+        }
```

Comment:
This test is currently failing, because we need to specify a different serializer in case the user is not the owner of the notebook, which does not show the url